### PR TITLE
Create a prow.viewer custom org role

### PR DESCRIPTION
**Pull request by spiffxp**
I want to empower others to see and troubleshoot whatever resources
constraints (or lack thereof) the release-blocking jobs are running under

Members of the k8s-infra-prow-viewers@kubernetes.io get this role on
projects related to prow

The role is a composite of:
- roles/compute.viewer
- roles/container.viewer
- roles/logging.viewer
- roles/monitoring.view

It's defined by a file and will be updated anytime
infra/gcp/ensure-e2e-projects.sh is run

I've currently manually created it and applied it to the
k8s-infra-prow-build project to verify access to expected resources

Members who have this role should be able to see two custom cloud
monitoring dashboards:
- https://console.cloud.google.com/monitoring/dashboards/custom/10510319052103514664?project=k8s-infra-prow-build&timeDomain=6h&authuser=1
- https://console.cloud.google.com/monitoring/dashboards/custom/10925237040785467832?authuser=1&project=k8s-infra-prow-build&timeDomain=6h

### CODE REVIEW

**Comment by spiffxp** [comment]
/cc @BenTheElder @liggitt 
In case you have suggestions on what else people should be able to see

**Comment by BenTheElder** [comment]
cluster view maybe?

File: infra/gcp/lib.sh [hunk 1]
```diff
@@ -610,3 +610,39 @@ function ensure_custom_iam_role() {
...
+    then
+      # be noisy when creating a role
+      gcloud iam roles create "${name}" --organization "${org}" --file "${file}"
+    else
+      # be quiet when updating, only output name of role
+      gcloud iam roles update "${name}" --organization "${org}" --file "${file}" | grep ^name:
+    fi
+}
+
+function custom_org_role_name() {
```
Comment (thockin) [inline infra/gcp/lib.sh:639]: Could do with comments

**Comment by dims** [comment]
@spiffxp i can confirm that i can see both dashboard! (had to drop the `authuser=1` from the url though)

**Comment by spiffxp** [comment]
>you have to expand premade roles like compute.viewer into their constituent parts

I wasn't thrilled by this either.  I didn't script creation of this role, which would be the next logical step.  Could periodically re-run that script to re-generate the role, so updates would propagate, but on a schedule of our own making.

I'll address the lack of comments in a followup, thanks.

**Comment by spiffxp** [comment]
Ran `./infra/gcp/ensure-e2e-projects.sh`

**Pull request merged**